src/main/java/atoml/smoke/SmoketestFromArff.java

```diff
@@ -27,7 +32,18 @@ public void generateData(int numFeatures, int numInstances, long seed) {
         catch (IOException e) {
             throw new RuntimeException("error reading ARFF from resource: " + resource, e);
         }
-        testdata = data;
+        if (additionalResource == null){
+			testdata = data;
+		} else{
+			InputStreamReader additionalFile = new InputStreamReader(
+					this.getClass().getResourceAsStream(additionalResource));
+			try(BufferedReader reader = new BufferedReader(additionalFile);) {
+				testdata = new Instances(reader);
+			}
+			catch (IOException e) {
+				throw new RuntimeException("error reading ARFF from resource: " + additionalResource, e);
+			}
+		}
```

Comment:
I would suggest to rather enforce that the resource cannot be null in the constructors and then loading both resources here, without a null check. This is minimally worse from a runtime perspective. The alternative would be to replace the reference assignment `testdata = data` with an deepcopy of data. 

Also: check the LGTM comment and see if you can fix the resource leak for additionalFile (I fully realize that I initially introduced this and you just copy&pasted this, so this is optional here)